### DISCUSSION TITLE
How do create a transformation between two rectangles.

### DISCUSSION THREAD

**Discussion by TormodS** [mono/SkiaSharp#2995] [Q&A]
In System.Drawing.Drawing2D.Matrix, the constructor accepted two rectangles as inputs and created the Matrix that would map any point from the one rectangle to the corresponding coordinate of the other. This was very useful in a number of cases, like when you wanted to do picture-in-picture combined with the fact that screen coordinates had 0,0 on the top left instead of bottom left (which often was the case for the source graphics.

I can't seem to find a constructor or factory for matrix that does this or an operation between two SKRect instances that return such a matrix. 

It seems a normal use case for a library dedicated to drawing artifacts on screen, so I am suspecting there is a whole toolbox somewhere that I haven't seen.

**Comment by mattleibow** [comment]
I don't think this exists. The reverse is true - given a matrix you can map any point to a new point.

However, this type of matrix constructor may be interesting to implement. Not sure if you would like to try and do that?
